Failing `TestWSClientNonBlockingEvents`

Ubuntu, go 1.19. Firstly discovered at https://github.com/nspcc-dev/neo-go/actions/runs/4883739029/jobs/8715507458?pr=3004.
```
2023-05-04T14:10:52.3767949Z === RUN   TestWSClientNonBlockingEvents
2023-05-04T14:10:52.3768321Z     wsclient_test.go:349: 
2023-05-04T14:10:52.3769041Z         	Error Trace:	/home/runner/work/neo-go/neo-go/pkg/rpcclient/wsclient_test.go:349
2023-05-04T14:10:52.3769525Z         	Error:      	Condition never satisfied
2023-05-04T14:10:52.3769973Z         	Test:       	TestWSClientNonBlockingEvents
2023-05-04T14:10:52.3770414Z --- FAIL: TestWSClientNonBlockingEvents (2.10s)
```

--- a/pkg/rpcclient/wsclient_test.go
+++ b/pkg/rpcclient/wsclient_test.go
@@ -155,11 +155,13 @@ func TestWSClientEvents(t *testing.T) {
 		fmt.Sprintf(`{"jsonrpc":"2.0","method":"block_added","params":[%s]}`, b1Verbose),
 		`{"jsonrpc":"2.0","method":"event_missed","params":[]}`, // the last one, will trigger receiver channels closing.
 	}
+	startSending := make(chan struct{})
 	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 		if req.URL.Path == "/ws" && req.Method == "GET" {
 			var upgrader = websocket.Upgrader{}
 			ws, err := upgrader.Upgrade(w, req, nil)
 			require.NoError(t, err)
+			<-startSending
 			for _, event := range events {
 				err = ws.SetWriteDeadline(time.Now().Add(2 * time.Second))
 				require.NoError(t, err)
@@ -209,6 +211,7 @@ func TestWSClientEvents(t *testing.T) {
 	// MissedEvent must close the channels above.
 
 	wsc.subscriptionsLock.Unlock()
+	close(startSending)
 
 	var (
 		b1Cnt, b2Cnt                                      int
@@ -297,11 +300,13 @@ func TestWSClientNonBlockingEvents(t *testing.T) {
 	require.True(t, chCap < len(events))
 
 	var blocksSent atomic.Bool
+	startSending := make(chan struct{})
 	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 		if req.URL.Path == "/ws" && req.Method == "GET" {
 			var upgrader = websocket.Upgrader{}
 			ws, err := upgrader.Upgrade(w, req, nil)
 			require.NoError(t, err)
+			<-startSending
 			for _, event := range events {
 				err = ws.SetWriteDeadline(time.Now().Add(2 * time.Second))
 				require.NoError(t, err)
@@ -331,6 +336,7 @@ func TestWSClientNonBlockingEvents(t *testing.T) {
 	wsc.receivers[chan<- *block.Block(bCh)] = []string{"0", "1"}
 	wsc.subscriptionsLock.Unlock()
 
+	close(startSending)
 	// Check that events are sent to WSClient.
 	require.Eventually(t, func() bool {
 		return blocksSent.Load()

